src/main/java/org/codehaus/groovy/transform/stc/StaticTypeCheckingVisitor.java

```diff
@@ -1834,19 +1838,22 @@ public void visitField(final FieldNode node) {
         try {
             typeCheckingContext.isInStaticContext = node.isInStaticContext();
             currentField = node;
-            super.visitField(node);
+            visitAnnotations(node);
```

Comment:
Normal visitor pattern would call super here. Perhaps a comment here to indicate that were replicating some of super.visitField to avoid processing init expression twice?

Reply:
If changed back, there are test cases that will fail.  It is technically possible to structure this so that the functional interface stuff comes before "super.visitField(node)" and the assignment type checking comes after (where it currently sits).

The property visit could have its initialization expression visit dropped, since it is redundant with the field.  I thought that would be a riskier change to make.